packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -272,20 +288,19 @@ protected override ConstructorProvider[] BuildConstructors()
             // handle sub-client constructors
             if (ClientOptionsParameter is null)
             {
-                _clientParameters = _subClientInternalConstructorParams;
```

Comment:
it is really a bad practice that the fact whether client parameters are initialized is bonded with constructors.
Actually, in the case of client parameters, we could fully get the list before any constructor is built. Therefore this line was removed.